xapian-letor/api/letor.cc

```diff
@@ -73,13 +72,13 @@ Letor::prepare_training_file(const string & query_file, const string & qrel_file
 }
 
 void
-Letor::create_ranker(int ranker_type, int metric_type) {
-    switch(ranker_type) {
-        case 0: internal->ranker = * new SVMRanker;
-                break;
-        case 1: break;
-        default: ;//cout<<"Please specify proper ranker.";
-    }
+Letor::create_ranker(int ranker_type, int metric_type) { //TODO: update when adding rankers
```

Comment:
@parthg certainly the CLI programs should have sensible defaults; the API will either (depending on what feels right) do something similar (as we do for instance with weighting schemes in core, which almost certainly will be appropriate for scorers), or you'll just instantiate whichever you need (which may make sense for rankers, depending on exactly what the API looks like).

Are you saying that the scorer default should be dependent on the ranker? That's obviously not a problem, and it may help settle the best direction for the API.